app/grandchallenge/serving/views.py

```diff
@@ -137,16 +137,16 @@ def serve_component_interface_value(
             pk=component_interface_value_pk
         )
     except (MultipleObjectsReturned, ComponentInterfaceValue.DoesNotExist):
-        raise Http404("No ComponentInterfaceValue found.")
-
-    if get_component_interface_values_for_user(
-        user=user, civ_pk=civ.pk
-    ).exists():
-        return protected_storage_redirect(
-            name=civ.file.name, creator=user, component_interface_value=civ
-        )
-
-    raise PermissionDenied
+        pass
+    else:
+        if get_component_interface_values_for_user(
+            user=user, civ_pk=civ.pk
+        ).exists():
+            return protected_storage_redirect(
+                name=civ.file.name, creator=user, component_interface_value=civ
+            )
+
+    raise Http404("No ComponentInterfaceValue found.")
```

Comment:
It's best not to rely on fall-through so we should be explicit with our denials, a 404 message is not necessary so we can omit it for consistency.

```suggestion
        raise Http404

    if get_component_interface_values_for_user(
        user=user, civ_pk=civ.pk
    ).exists():
        return protected_storage_redirect(
            name=civ.file.name, creator=user, component_interface_value=civ
        )

    raise Http404
```

Reply:
Good to know!